aredis_om/_compat.py

```diff
@@ -1,19 +1,99 @@
+from dataclasses import dataclass, is_dataclass
+from typing import (
+    Any,
+    Callable,
+    Deque,
+    Dict,
+    FrozenSet,
+    List,
+    Mapping,
+    Sequence,
+    Set,
+    Tuple,
+    Type,
+    Union,
+)
+
 from pydantic.version import VERSION as PYDANTIC_VERSION
+from typing_extensions import Annotated, Literal, get_args, get_origin
 
 
 PYDANTIC_V2 = PYDANTIC_VERSION.startswith("2.")
```

Comment:
If it's helpful, I have seen some shim implementations that just use try except.. but this may not necessarily be better. Just offering some other ways to approach this:

https://github.com/langchain-ai/langchain/blob/master/libs/core/langchain_core/pydantic_v1/dataclasses.py

Reply:
I think this is more or less the way that [FastAPI](https://github.com/tiangolo/fastapi/blob/master/fastapi/_compat.py#L27) does it. a try/except might be a bit more inclusive. . . hmm thoughts @banker or @bsbodden ?

Reply:
The try/except approach makes me feel icky. Are we thinking of dropping Pydantic 1.0 support completely? It's seems a lot of project are doing that nowadays.